pages/content/amp-dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cache-urls.md

```diff
@@ -0,0 +1,75 @@
+---
+$title: AMP Cache URL Format and Request Handling
+$order: 9
+toc: false
+formats:
+  - websites
+  - stories
+  - ads
+---
+
+In this document, you'll learn about the AMP Cache URL format and how it handles requests. 
+
+## URL Format
+
+When possible, the Google AMP Cache will create a subdomain for each AMP document's domain by first converting it from [IDN (punycode)](https://en.wikipedia.org/wiki/Punycode) to UTF-8. The caches replaces every `-` (dash) with `--` (2 dashes) and replace every `.` (dot) with `-` (dash). For example, `pub.com` will map to `pub-com.cdn.ampproject.org`.
+
+You can use this URL calculator to convert an URL into an AMP cache version:
+
+<div>
+<amp-iframe title="AMP Cache tool" height="104" layout="fixed-height" sandbox="allow-scripts allow-same-origin allow-popups" src="{{ doc.pod.podspec.base_urls.preview }}/static/samples/files/amp-url-converter.html?url=https://amp.dev/index.amp.html">
+  <div placeholder></div>
+</amp-iframe>
+</div>
+
+The converted AMP Cache URL consists of the following parts:
+
+* `amp-dev`: the publisher domain converted into a subdomain using the algorithm described above.</li>
+* `cdn.ampproject.org`: the AMP Cache domain.
+* `c`: to indicate that it's an AMP document (there is also `i` for image and `r` for resources such as fonts).
+* `s`: indicating that the AMP Cache fetches the content from the origin using TLS (secure HTTPS).
+* `amp-dev.com/index.amp.html`: the original URL excluding the scheme.
+
+It is OK for URLs to include parameters in the query string, simply include these in the AMP Cache URL as well.
+
+Example: [https://amp-dev.cdn.ampproject.org/c/s/amp.dev/documentation/examples/api/query?value=Hello%20World](https://amp-dev.cdn.ampproject.org/c/s/amp.dev/documentation/examples/api/query?value=Hello%20World).
+
+## Redirect & Error Handling
+
+Here are some examples for how the AMP Cache handles redirects and errors:
+
+**Redirects**
+
+The AMP Cache follows redirects when resolving AMP URLs. For example, if an URL redirects to another AMP URL:
+
+```
+$ curl -I https://amp.dev/documentation/examples/api/redirect?url=https://amp.dev/index.amp.html
+HTTP/1.1 301 Moved Permanently
+Content-Type: text/html; charset=utf-8
+Location: https://amp.dev/index.amp.html
+...
+```
+
+Then the AMP Cache will return the content of the resolved redirect for the original URL.
+
+Example: [https://amp-dev.cdn.ampproject.org/amp.dev/documentation/examples/api/redirect?url=https://amp.dev/index.amp.html](https://amp-dev.cdn.ampproject.org/amp.dev/documentation/examples/api/redirect?url=https://amp.dev/index.amp.html).
+
+Important: If you move the location of the AMP Files on your server, make sure to set up a redirect from the old location to the new one.
```

Comment:
Should we wrap this in an important block?
`[tip type="important"]...[/tip]`

Reply:
that happens automatically if you start a paragraph with `Important:` or `Note: ` or ...

Reply:
Nifty - thanks!